AK/StringUtils.h

```diff
@@ -38,6 +39,13 @@ enum class TrimWhitespace {
     No,
 };
 
+enum class OnSplit : unsigned {
+    Nothing = 0,
```

Comment:
These all need a small comment explaining what each does please. `Nothing` is not very clear by itself.

Reply:
`Nothing` does... nothing. To indicate it should neither keep empty or keep separators. What should be said?

Reply:
I added some comments

Reply:
> To indicate it should neither keep empty or keep separators. What should be said?

Just that. If it actually were doing nothing, I would expect an empty results vector :)

Reply:
done :^)